common/src/main/java/com/wynntils/services/mapdata/attributes/type/MapVisibility.java

```diff
@@ -4,12 +4,12 @@
  */
 package com.wynntils.services.mapdata.attributes.type;
 
-public enum MapVisibility {
-    ALWAYS,
-    NEVER
-
-    // FIXME: This needs to be more complex, and contain (at least):
-    // below zoom: first visible at, fully visible at
-    // above zoom: fully visible at, last visible at
-    // between zoom: both above
+@FunctionalInterface
+public interface MapVisibility {
+    /**
+     * This method is used to determine the visibility of a feature at a given zoom level.
+     * @param zoomStep The current zoom step of the map. You can read more about zoom steps in the {@link com.wynntils.utils.render.MapRenderer} class.
+     * @return A float between 0 and 1, where 0 means the feature is not visible and 1 means the feature is fully visible. Values in between are used to determine the transparency of the feature.
+     */
+    float getVisibility(int zoomStep);
```

Comment:
Given how the zoom PR ended up, we should use the term "zoom level" and have it as a float everywhere.

```suggestion
    float getVisibility(float zoomLevel);
```